portable_concurrency/bits/once_consumable_stack.h

```diff
@@ -25,22 +25,23 @@ struct forward_list_node {
   forward_list_node* next;
 };
 
-template<typename T>
-void forward_list_deleter<T>::operator() (forward_list_node<T>* head) noexcept {
+template<typename T, typename Alloc>
+void forward_list_deleter<T, Alloc>::operator() (forward_list_node<T>* head) noexcept {
   if (!head)
     return;
   for (auto* p = head; p != nullptr;) {
     auto* to_delete = p;
     p = p->next;
-    delete to_delete;
+    std::allocator_traits<Alloc>::destroy(allocator, to_delete);
+    std::allocator_traits<Alloc>::deallocate(allocator, to_delete, 1);
   }
 }
 
-template<typename T>
+template<typename T, typename Alloc>
```

Comment:
Iterator do not owns items and do not need to be allocator aware. Looks like the only place which requires information about allocator type is the following constructor `forward_list_iterator(forward_list<T, Alloc>& list)`. It would be better to make only this constructor allocator aware.